snuba/util.py

```diff
@@ -291,7 +291,7 @@ def conditions_expr(dataset, conditions, body, depth=0):
         # (IN, =, LIKE) are looking for rows where any array value matches, and
         # exclusionary operators (NOT IN, NOT LIKE, !=) are looking for rows
         # where all elements match (eg. all NOT LIKE 'foo').
-        columns = dataset.get_schema().get_columns()
+        columns = dataset.get_read_schema().get_columns()
```

Comment:
I think it would be better to pass the schema through instead of the dataset to this whole file so we reduce the coupling between these functions and the read path which is not necessary.

Reply:
Hmmm, I'll try doing this.